solution1/solution1.go

```diff
@@ -0,0 +1,180 @@
+package main
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"os/user"
+	"path"
+	"strings"
+)
+
+func init() {
+	log.SetFlags(log.Ldate | log.Ltime | log.LUTC | log.Lshortfile)
+	log.SetPrefix("I ")
+}
+
+type Config struct {
+	K8sServer  string
+	K8sCrtFile string
+	K8sKeyFile string
+	K8sCAFile  string
+}
+
+func main() {
+	config := parseFlags()
+
+	err := mainError(config)
+	if err != nil {
+		log.SetPrefix("E ")
+		log.Printf("%s", err)
+		os.Exit(1)
+	}
+}
+
+func parseFlags() Config {
+	var config Config
+
+	var homedir string
+	{
+		u, err := user.Current()
+		if err != nil {
+			homedir = os.Getenv("HOME")
+		} else {
+			homedir = u.HomeDir
+		}
+
+	}
+
+	var serverDefault string
+	{
+		out, err := exec.Command("minikube", "ip").Output()
+		if err == nil {
+			minikubeIP := strings.TrimSpace(string(out))
+			serverDefault = "https://" + string(minikubeIP) + ":8443"
+		}
+	}
+
+	flag.StringVar(&config.K8sServer, "kubernetes.server", serverDefault, "Kubernetes API server address.")
+	flag.StringVar(&config.K8sCrtFile, "kubernetes.crt", path.Join(homedir, ".minikube/apiserver.crt"), "Kubernetes certificate file path.")
+	flag.StringVar(&config.K8sKeyFile, "kubernetes.key", path.Join(homedir, ".minikube/apiserver.key"), "Kubernetes key file path.")
+	flag.StringVar(&config.K8sCAFile, "kubernetes.ca", path.Join(homedir, ".minikube/ca.crt"), "Kubernetes CA file path.")
+	flag.Parse()
+
+	return config
+}
+
+func mainError(config Config) error {
+	k8sClient, err := newHttpClient(config)
+	if err != nil {
+		return err
+	}
+
+	// Create Custom Resource Definition.
+	{
+		log.Printf("creating custom resource")
+
+		res, err := k8sClient.Post(
+			config.K8sServer+"/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions",
```

Comment:
A constant here would be good. At the end we can show how operatorkit helps with paths.

Reply:
I agree, but I don't want ppl to jump around, I'd like them to see immediately the path. We shouldn't consider it production grade code. WDYT?

Reply:
Good point, no need for a constant.